internal/cutil/ptrguard_test.go

```diff
@@ -0,0 +1,66 @@
+package cutil
+
+import (
+	"math/rand"
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPtrGuard(t *testing.T) {
+	t.Run("storeAndRelease", func(t *testing.T) {
+		s := "string"
+		goPtr := (unsafe.Pointer)(&s)
+		cPtr := Malloc(PtrSize)
+		defer Free(cPtr)
+		pg := NewPtrGuard(cPtr, goPtr)
+		assert.Equal(t, *(*unsafe.Pointer)(cPtr), goPtr)
+		pg.Release()
+		assert.Zero(t, *(*unsafe.Pointer)(cPtr))
+	})
+
+	t.Run("multiRelease", func(t *testing.T) {
+		s := "string"
+		goPtr := (unsafe.Pointer)(&s)
+		cPtr := Malloc(PtrSize)
+		defer Free(cPtr)
+		pg := NewPtrGuard(cPtr, goPtr)
+		assert.Equal(t, *(*unsafe.Pointer)(cPtr), goPtr)
+		pg.Release()
+		pg.Release()
+		pg.Release()
+		pg.Release()
+		assert.Zero(t, *(*unsafe.Pointer)(cPtr))
+	})
+
+	t.Run("stressTest", func(t *testing.T) {
+		const N = 20000 // assert that no threads are created (default limit 10000)
+		const M = 100000
+		var ptrGuards [N]*PtrGuard
+		cPtrArr := (*[N]CPtr)(unsafe.Pointer(Malloc(N * PtrSize)))
+		defer Free(CPtr(&cPtrArr[0]))
+		for n := 0; n < M; n++ {
+			i := uintptr(rand.Intn(N))
```

Comment:
Because you use a random index in the list of PtrGuards, there is no guarantee there will be > 10000 instances? I think that makes this test unreliable...

What if `rand.Intn(N)` always returns the same value? Some test frameworks seem to prevent real randomness. Although I do not assume it is the case with Go.

Reply:
Yeah, good catch, but this is not the case here. If not seeded with some entropy, the random numbers are deterministically random, that means, if the tests succeeds one time, it will always succeed. (At least, that was the case, when I dealt with the random seeding in heketi long time ago, didn't check again today.) But I agree, it's better if I fill the whole array first, and then start the random loop. Will update accordingly.